### PR TITLE
Updated PubSub docs, removing unused types + top-level comments

### PR DESCRIPTION
**Pull request by TheIronDev**
Updated PubSub docs, removing unused types + top-level comments

### CODE REVIEW

File: src/v2/providers/pubsub.ts [hunk 1]
```diff
@@ -4,20 +4,25 @@ import { CloudEvent, CloudFunction } from '../core';
 import * as options from '../options';
 
 /**
- * A PubSub Topic is:
+ * Google Cloud's Pub/Sub is a globally distributed message bus that automatically scales as you need it.
```
Comment (joehan) [inline src/v2/providers/pubsub.ts:7]: I think this should be just `Google Cloud Pub/Sub`

File: src/v2/providers/pubsub.ts [hunk 2]
```diff
@@ -4,20 +4,25 @@ import { CloudEvent, CloudFunction } from '../core';
...
+ * You can create a function ({@link onMessagePublished}) that handles Pub/Sub events by using functions.pubsub.
+ *
+ * You can trigger a function whenever a new Pub/Sub message is sent to a specific topic.
```
Comment (joehan) [inline src/v2/providers/pubsub.ts:10]: Nit: 'This functions triggers whenever a new Pub/Sub...`

**Review by joehan — APPROVED** [review]
Some nits, but I see that this is copied from the linked docs, so feel free to ignore

File: src/v2/providers/pubsub.ts (hunk 1 above)
Comment (inlined) [reply src/v2/providers/pubsub.ts:7]: +1

**Review by inlined — APPROVED** [review]
Great docs!

File: src/v2/providers/pubsub.ts (hunk 2 above)
Comment (TheIronDev) [reply src/v2/providers/pubsub.ts]: Makes sense, updated!

**Pull request merged**